addons/event/models/event.py

```diff
@@ -390,11 +390,15 @@ def _onchange_partner(self):
     @api.multi
     def message_get_suggested_recipients(self):
         recipients = super(EventRegistration, self).message_get_suggested_recipients()
+        public_users = self.env["res.users"]
```

Comment:
I would use list of user ids to compare them instead of comparing record sets. When playing with users I think it is better to compare id list to avoid access rights issues.
```
public_group = self.env.ref('base.group_public')
if public_group:
    public_user_ids = public_group.sudo().mapped('users').ids
else:
    public_user_ids = []
```